### PR TITLE
Adding uuid to namespace for index to make unique in ACL, maintaining…

### PR DESCRIPTION
**Pull request by ewolinetz**
… formatting within kibana

@sosiouxme @jcantrill PTAL

### CODE REVIEW

File: src/main/java/io/fabric8/elasticsearch/plugin/acl/DynamicACLFilter.java [hunk 1]
```diff
@@ -196,12 +195,22 @@ private boolean updateCache(final String user, final String token, final boolean
...
+				if ( ! isBlacklistProject(project.getMetadata().getName()) )
+					names.add(project.getMetadata().getName() + "." + project.getMetadata().getUid());
 			}
 		}
 		return names;
 	}
 	
+	private boolean isBlacklistProject(String project) {
+		
+		for ( String blacklist : BLACKLIST_PROJECTS )
```
Comment (jcantrill) [inline src/main/java/io/fabric8/elasticsearch/plugin/acl/DynamicACLFilter.java]: ArraysUtils.contains(BLACKLIST_PROJECTS, project.toLowerCase());

File: src/main/java/io/fabric8/elasticsearch/plugin/kibana/KibanaSeed.java [hunk 2]
```diff
@@ -105,12 +98,30 @@ public static void setDashboards(String user, Set<String> projects, Set<String>
 		else {
 			List<String> common = new ArrayList<String>(indexPatterns);
 			
+			// build normalizedSort
+			List<String> normalizedSort = new ArrayList<String>();
```
Comment (jcantrill) [inline src/main/java/io/fabric8/elasticsearch/plugin/kibana/KibanaSeed.java]: Can actually be simplified to:

List<String> normalizedSort = new ArrayList<>();

**Comment by jcantrill** [comment]
Add tests please.

File: src/main/java/io/fabric8/elasticsearch/plugin/kibana/KibanaSeed.java (hunk 2 above)
Comment (ewolinetz) [inline src/main/java/io/fabric8/elasticsearch/plugin/kibana/KibanaSeed.java]: Don't I need to then cast everything later on, such as with line 117?  Not sure if we gain anything by doing this...

Comment (jcantrill) [inline src/main/java/io/fabric8/elasticsearch/plugin/kibana/KibanaSeed.java]: Actually, I wrote my suggestion incorrectly.  It should have been:

List<String> normalizedSort = new ArrayList<>();

Its just a shortcut where you need to type less.

Comment (jimmidyson) [inline src/main/java/io/fabric8/elasticsearch/plugin/kibana/KibanaSeed.java]: Diamond operator ftw! Generic types inferred from declaration. Look out for more inference to come in Java in the future hopefully, e.g. proposed http://openjdk.java.net/jeps/286

**Comment by ewolinetz** [comment]
This is to address https://bugzilla.redhat.com/show_bug.cgi?id=1316216

**Comment by ewolinetz** [comment]
There seems to be an issue with Kibana being able to msearch for a non-cluster admin when the ACL doesn't grant access to indices in the format `project.YYYYMMDD`.  Might need Kibana to display the uuid for the project which was trying to avoid

**Comment by jcantrill** [comment]
Reference https://github.com/kubernetes/kubernetes/issues/23001

**Comment by jimmidyson** [comment]
retest this please

**Comment by jimmidyson** [comment]
@jcantrill Sorry about the test weirdness, bad Jenkins setup... can't get the staff these days.

**Comment by ewolinetz** [comment]
[test]

**Comment by sosiouxme** [comment]
@ewolinetz are there corresponding fluentd changes? Maybe I'm just missing it but I don't see anywhere that incoming data is being re-indexed so I assume fluentd must be using the new index format too...

**Comment by sosiouxme** [comment]
Argh, finger slipped. Never mind all that, obviously fluentd changes would be in the other repo...

**Comment by ewolinetz** [comment]
@sosiouxme yeah, there will be fluentd changes that need to go along with this

**Comment by sosiouxme** [comment]
so i guess "LGTM" too

**Comment by ewolinetz** [comment]
squashed commits

**Comment by jcantrill** [comment]
lgtm

**Comment by ewolinetz** [comment]
[merge]

**Pull request merged**